From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 22 Aug 2017 22:04:28 +0200
Subject: [PATCH] states/maas: Add mcp.rsa.pub to authorized_keys

Add our mcp.rsa.pub RSA key to all nodes, including VCP VMs.
This is required for functest to be able to fetch openrc.

While at it, add retry wrappers for more VCP VM state.sls calls.

Change-Id: I34f79848c52e36de8d981055880321a081420874
Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Signed-off-by: Guillermo Herrero <Guillermo.Herrero@enea.com>
---
 mcp/config/states/maas | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/mcp/config/states/maas b/mcp/config/states/maas
index 52a9b77..600578f 100755
--- a/mcp/config/states/maas
+++ b/mcp/config/states/maas
@@ -69,6 +69,9 @@ while [ $rc -ne 0 ]; do
   sleep 5
 done

-salt -C '* and not cfg01* and not mas01*' saltutil.sync_all
-salt -C '* and not cfg01* and not mas01*' state.apply salt
+wait_for "salt -C '* and not cfg01* and not mas01*' ssh.set_auth_key ${SUDO_USER} \
+  $(awk 'NR==1{print $2}' $(eval echo ~${SUDO_USER}/.ssh/authorized_keys))"
+
+wait_for "salt -C '* and not cfg01* and not mas01*' saltutil.sync_all"
+wait_for "salt -C '* and not cfg01* and not mas01*' state.apply salt"
 wait_for "salt -C '* and not cfg01* and not mas01*' state.apply linux,ntp"
